src/psyclone/tests/psyir/transformations/intrinsics/sum2code_trans_test.py

```diff
@@ -37,513 +37,137 @@
-        "  integer :: i_1\n"
-        "  integer :: i_2\n\n"
-        "  sum_var = 0.0\n"
-        "  do i_2 = LBOUND(array, 3), UBOUND(array, 3), 1\n"
-        "    do i_1 = LBOUND(array, 2), UBOUND(array, 2), 1\n"
-        "      do i_0 = LBOUND(array, 1), UBOUND(array, 1), 1\n"
-        "        sum_var(i_0,i_2) = sum_var(i_0,i_2) + array(i_0,i_1,i_2)\n"
-        "      enddo\n"
-        "    enddo\n"
-        "  enddo\n"
-        "  result(:,:) = value1 + sum_var * value2\n\n"
-        "end subroutine sum_test\n")
-    psyir = fortran_reader.psyir_from_source(code)
-    # FileContainer/Routine/Assignment/BinaryOperation(ADD)/
-    # BinaryOperation(MUL)/UnaryOperation
-    sum_node = psyir.children[0].children[0].children[1].children[1]. \
-        children[0]
-    trans = Sum2CodeTrans()
-    trans.apply(sum_node)
-    result = fortran_writer(psyir)
-    assert result == expected
-
-
-# specified array range
-def test_apply_dimension_multid_range(fortran_reader, fortran_writer):
-    '''Test that the apply method works as expected when an array range is
-    specified and the array is multi-dimensional. Only the specified
-    dimension should be summed.
-
-    '''
-    code = (
-        "subroutine sum_test(array,value1,value2,n,m,p)\n"
-        "  integer :: n,m,p\n"
-        "  real :: array(:,:,:)\n"
-        "  real :: value1, value2\n"
-        "  real :: result(n,p)\n"
-        "  result(:,:) = value1 + sum(array(1:n,m-1:m,1:p),dim=2) * "
-        "value2\n"
-        "end subroutine\n")
-    expected = (
-        "subroutine sum_test(array, value1, value2, n, m, p)\n"
-        "  integer :: n\n  integer :: m\n  integer :: p\n"
-        "  real, dimension(:,:,:) :: array\n"
-        "  real :: value1\n  real :: value2\n"
-        "  real, dimension(n,p) :: result\n"
-        "  real, dimension(n,p) :: sum_var\n"
-        "  integer :: i_0\n  integer :: i_1\n  integer :: i_2\n\n"
-        "  sum_var = 0.0\n"
-        "  do i_2 = 1, p, 1\n"
-        "    do i_1 = m - 1, m, 1\n"
-        "      do i_0 = 1, n, 1\n"
-        "        sum_var(i_0,i_2) = sum_var(i_0,i_2) + array(i_0,i_1,i_2)\n"
-        "      enddo\n"
+        "  do i_1 = 1, 20, 1\n"
+        "    do i_0 = 1, 10, 1\n"
+        "      sum_var = sum_var + array(i_0,i_1)\n"
         "    enddo\n"
         "  enddo\n"
-        "  result(:,:) = value1 + sum_var * value2\n\n"
+        "  result = sum_var\n\n"
         "end subroutine sum_test\n")
     psyir = fortran_reader.psyir_from_source(code)
-    # FileContainer/Routine/Assignment/BinaryOperation(ADD)/
-    # BinaryOperation(MUL)/UnaryOperation
-    sum_node = psyir.children[0].children[0].children[1].children[1]. \
-        children[0]
+    # FileContainer/Routine/Assignment/UnaryOperation
+    intrinsic_node = psyir.children[0].children[0].children[1]
     trans = Sum2CodeTrans()
-    trans.apply(sum_node)
+    trans.apply(intrinsic_node)
     result = fortran_writer(psyir)
     assert result == expected
```

Comment:
You've guessed it...compilation check please.

Reply:
Added.